morebits.js

```diff
@@ -3298,6 +3267,63 @@ Morebits.wiki.page = function(pageName, currentAction) {
 
 	};
 
+	// Common checks for action methods
+	// Used for move, undelete, delete, protect, stabilize
+	var fnPreflightChecks = function(action, onFailure) {
+		// if a non-admin tries to do this, don't bother
+		if (!Morebits.userIsSysop && action !== 'move') {
+			ctx.statusElement.error('Cannot ' + action + 'page : only admins can do that');
+			onFailure(this);
+			return false;
+		}
+
+		if (!ctx.editSummary) {
+			ctx.statusElement.error('Internal error: ' + action + ' reason not set (use setEditSummary function)!');
+			onFailure(this);
+			return false;
+		}
+		return true; // all OK
+	};
+
+	// Common checks for fnProcess functions (fnProcessDelete, fnProcessMove, etc.)
+	// Used for move, undelete, delete, protect, stabilize
+	var fnProcessChecks = function(action, onFailure, xml) {
+		var missing = $(xml).find('page').attr('missing') === '';
+		if ((action === 'undelete' && !missing) ||
+			(action === 'protect' && (
+				(!missing && ctx.protectCreate) ||
+					(missing && (ctx.protectEdit || ctx.protectMove)))) ||
+			missing) {
+			ctx.statusElement.error('Cannot ' + action + ' the page because it ' + (missing ? 'no longer' : 'already') + ' exists');
+			onFailure(this);
+			return false;
+		}
+
+		// Delete, undelete, move
+		// extract protection info
+		var editprot;
+		if (action === 'undelete') {
+			editprot = $(xml).find('pr[type="create"]');
+		} else if (action === 'delete' || action === 'move') {
+			editprot = $(xml).find('pr[type="edit"]');
+		}
+		if (editprot.length > 0 && editprot.attr('level') === 'sysop' && !ctx.suppressProtectWarning &&
+			!confirm('You are about to ' + action + ' the fully protected page "' + ctx.pageName +
+			(editprot.attr('expiry') === 'infinity' ? '" (protected indefinitely)' : '" (protection expiring ' + new Morebits.date(editprot.attr('expiry')).calendar('utc') + ' (UTC))') +
+			'.  \n\nClick OK to proceed with ' + action + ', or Cancel to skip.')) {
+			ctx.statusElement.error('Aborted ' + action + ' on fully protected page.');
```

Comment:
Bad wording on stabilize: `action` here is `apply FlaggedRevs to` which doesn't fit.

Reply:
Constant struggle with FlaggedRevs, Pending Changes, and Stabilize all being awful, inscrutable, or not verbs!